test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/ExpressionRewriterTest.cs

```diff
@@ -19,6 +19,37 @@ namespace Microsoft.AspNet.Mvc.Razor.Compilation
     public class ExpressionRewriterTest
     {
         [Fact]
+        public void ExpressionRewriter_DoesNotThrowsOnUnknownTypes()
+        {
+            // Arrange
+            var source = @"
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.Razor;
+
+public class ExamplePage : RazorPage
+{
+    public IViewComponentHelper Component { get; set; }
+
+    public override async Task ExecuteAsync()
+    {
+        Write(
+            await Component.InvokeAsync(
+                ""SomeComponent"",
+                item => new HelperResult((__razor_template_writer) => WriteLiteralTo(__razor_template_writer, ""Hello World""))));
+        }
+    }
+";
+            var tree = CSharpSyntaxTree.ParseText(source);
+            var compilation = Compile(tree, allowErrors: true);
```

Comment:
Exactly how does the system get into a state where it ignores C# compilation errors? That is, how is the `ExpressionRewriter` reached after compilation fails?

I'm wondering whether there's a more important underlying bug that this PR partially works around.